### PR TITLE
CRYP-69 : Added Endpoint for ArtistWalletAddress

### PR DESCRIPTION
**Pull request by Afaqsoftaxus**
## Jira

- https://softaxus-jira.atlassian.net/browse/CRYP-69

## Changes

- Added Endpoint For GetArtistByWalletAddress

## Testing

- https://softaxus-jira.atlassian.net/browse/CRYP-69?focusedCommentId=10108

### CODE REVIEW

File: src/CryptoAxus.API/Controllers/ArtistController.cs [hunk 1]
```diff
@@ -122,6 +124,29 @@ public async Task<IActionResult> PostArtist([FromBody] CreateArtistDto artist)
             return BadRequest(response);
         return CreatedAtRoute("GetArtistById", new { id = response.Result?.Id }, response);
     }
```
Comment (junaid-softaxus) [inline src/CryptoAxus.API/Controllers/ArtistController.cs:126]: Add blank space at the end method end

File: src/CryptoAxus.API/Controllers/ArtistController.cs [hunk 2]
```diff
@@ -122,6 +124,29 @@ public async Task<IActionResult> PostArtist([FromBody] CreateArtistDto artist)
...
+    /// </summary>
```
Comment (junaid-softaxus) [inline src/CryptoAxus.API/Controllers/ArtistController.cs:127]: This should be summary opening tag `<summary>`

File: src/CryptoAxus.API/Controllers/ArtistController.cs [hunk 3]
```diff
@@ -122,6 +124,29 @@ public async Task<IActionResult> PostArtist([FromBody] CreateArtistDto artist)
...
+    /// <param name="userWalletAddress" example="0x507f191e810c19729de860ea"></param>
+    /// <response code="200">Success response with 200 code and information message about update</response>
+    /// <response code="404">Not Found response with 404 code and information message</response>
+    /// <response code="400">Bad Request response with 400 code and information message</response>
+    /// <returns></returns>
```
Comment (junaid-softaxus) [inline src/CryptoAxus.API/Controllers/ArtistController.cs:132]: Add summary closing tag `</summary>` after `<returns>` tag

File: src/CryptoAxus.API/Controllers/ArtistController.cs [hunk 4]
```diff
@@ -122,6 +124,29 @@ public async Task<IActionResult> PostArtist([FromBody] CreateArtistDto artist)
...
+    [HttpGet("{userWalletAddress:required}/userWalletAddress", Name = "GetArtistByWalletAddressRequest", Order = 5)]
```
Comment (junaid-softaxus) [inline src/CryptoAxus.API/Controllers/ArtistController.cs:133]: `userWalletAddress` data type missing in route template

File: src/CryptoAxus.API/Controllers/ArtistController.cs [hunk 5]
```diff
@@ -162,6 +187,13 @@ private IReadOnlyList<Links> CreateArtistLinks(ArtistDto dto, string? fields)
                                        $"{HttpContext?.Request.Scheme}://{HttpContext?.Request.Host}{Constants.ApiValue}");
         links.Add(link);
 
+        link = new Links(Url.RouteUrl("GetArtistByWalletAddress", new { dto.UserWalletAddress }),
+                         "artist_walletAddress",
```
Comment (junaid-softaxus) [inline src/CryptoAxus.API/Controllers/ArtistController.cs:191]: This value should be `get_userWalletAddress`

File: src/CryptoAxus.API/Controllers/RootController.cs [hunk 6]
```diff
@@ -33,7 +33,11 @@ public async Task<ActionResult<BaseResponse<IReadOnlyList<Links>>>> GetRoot()
 
                 new Links(href: Url.RouteUrl("PostArtist"),
                           rel: "post",
-                          method: Constants.PostMethod)
+                          method: Constants.PostMethod),
+
+               new Links(href: Url.RouteUrl(routeName: "GetArtistByWalletAddress", values: new { userWalletAddress = "0x647115d2b38bc8ea242beb01" }),
+                          rel: "artist",
```
Comment (junaid-softaxus) [inline src/CryptoAxus.API/Controllers/RootController.cs:39]: Rel field should be get_userWalletAddress

File: src/CryptoAxus.Application/Features/Artist/GetArtistByWalletAddress/Handler/GetArtistDataByWalletAddress.cs [hunk 7]
```diff
@@ -0,0 +1,33 @@
+﻿using CryptoAxus.Application.Features.Artist.GetArtistByWalletAddress.Request;
+using CryptoAxus.Application.Features.Artist.GetArtistByWalletAddress.Response;
```
Comment (junaid-softaxus) [inline src/CryptoAxus.Application/Features/Artist/GetArtistByWalletAddress/Handler/GetArtistDataByWalletAddress.cs:2]: Using statements should be extracted to the relevant GlobalUsing file

File: src/CryptoAxus.Application/Features/Artist/GetArtistByWalletAddress/Handler/GetArtistDataByWalletAddress.cs [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+{
+    private readonly IRepository<ArtistDocument> _repository;
+
+    public GetArtistByWalletAddressHandler(IRepository<ArtistDocument> repository, ILogger<GetArtistByWalletAddressHandler> logger) : base(logger)
+    {
+        _repository = repository;
+    }
+
+    public async Task<GetArtistByWalletAddressResponse> Handle(GetArtistByWalletAddressRequest request,
+                                                      CancellationToken cancellationToken = default)
```
Comment (junaid-softaxus) [inline src/CryptoAxus.Application/Features/Artist/GetArtistByWalletAddress/Handler/GetArtistDataByWalletAddress.cs:16]: Parameters alignment not correct.

File: src/CryptoAxus.Application/Features/Artist/GetArtistByWalletAddress/Handler/GetArtistDataByWalletAddress.cs [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+        if (artist is null)
+            return new GetArtistByWalletAddressResponse(HttpStatusCode.NotFound,
+                                             message: $"No artist found against WalletAddress: {request.UserWalletAddress}",
+                                             result: null);
+
+        ArtistDto artistDto = artist.Adapt<ArtistDto>();
+
+        return new GetArtistByWalletAddressResponse(HttpStatusCode.OK,
+                                         message: "Artist record found successfully",
+                                         result: artistDto);
```
Comment (junaid-softaxus) [inline src/CryptoAxus.Application/Features/Artist/GetArtistByWalletAddress/Handler/GetArtistDataByWalletAddress.cs:31]: Parameters not aligned correctly.

File: src/CryptoAxus.Application/Features/Artist/GetArtistByWalletAddress/Request/GetArtistByWalletAddressRequest.cs [hunk 10]
```diff
@@ -0,0 +1,14 @@
+﻿using CryptoAxus.Application.Features.Artist.GetArtistByWalletAddress.Response;
```
Comment (junaid-softaxus) [inline src/CryptoAxus.Application/Features/Artist/GetArtistByWalletAddress/Request/GetArtistByWalletAddressRequest.cs:1]: See my above comments

File: src/CryptoAxus.Application/Features/Artist/GetArtistByWalletAddress/Response/NotFoundArtistByWalletAddressResponseExample.cs [hunk 11]
```diff
@@ -0,0 +1,20 @@
+﻿using CryptoAxus.Common.Services.Contracts;
```
Comment (junaid-softaxus) [inline src/CryptoAxus.Application/Features/Artist/GetArtistByWalletAddress/Response/NotFoundArtistByWalletAddressResponseExample.cs:1]: Add using in the GlobalUsing

File: test/CryptoAxus.Application.Features.Artist/GetArtistByWalletAddress/GetArtistByWalletAddressTestData.cs [hunk 12]
```diff
@@ -0,0 +1,64 @@
+﻿using CryptoAxus.Application.Features.Artist.GetArtistByWalletAddress.Handler;
+using CryptoAxus.Application.Features.Artist.GetArtistByWalletAddress.Request;
+using CryptoAxus.Domain.Collections;
+using Moq;
```
Comment (junaid-softaxus) [inline test/CryptoAxus.Application.Features.Artist/GetArtistByWalletAddress/GetArtistByWalletAddressTestData.cs:4]: See my previous comments

File: test/CryptoAxus.Application.Features.Artist/GetArtistByWalletAddress/Handler/GetArtistByWalletAddressHandlerTests.cs [hunk 13]
```diff
@@ -0,0 +1,55 @@
+﻿namespace CryptoAxus.Application.Features.Artist.GetArtistByWalletAddress.Handler;
+
+public class GetArtistByWalletAddressHandlerTests : GetArtistByWalletAddressTestData
+{
+    [Fact]
+    public async Task When_Artist_WalletAddress_Is_Passed_Expect_Artist_Document_Is_Returned()
```
Comment (junaid-softaxus) [inline test/CryptoAxus.Application.Features.Artist/GetArtistByWalletAddress/Handler/GetArtistByWalletAddressHandlerTests.cs:6]: Add _ between WalletAddress, should be Wallet_Address

File: test/CryptoAxus.Application.Features.Artist/GetArtistByWalletAddress/Handler/GetArtistByWalletAddressHandlerTests.cs [hunk 14]
```diff
@@ -0,0 +1,55 @@
...
+        var response = await sut.Handle(request);
+
+        // Assert
+        response.StatusCode.ShouldBe(HttpStatusCode.OK);
+        response.IsSuccessful.ShouldBe(true);
+        response.Result?.Username.ShouldBe("testUsername");
+    }
+
+    [Fact]
+    public async Task When_Artist_WalletAddress_Is_Passed_Expect_Not_Found_Result()
```
Comment (junaid-softaxus) [inline test/CryptoAxus.Application.Features.Artist/GetArtistByWalletAddress/Handler/GetArtistByWalletAddressHandlerTests.cs:23]: See my comment above.

**Review by junaid-softaxus — COMMENTED** [review]
Update the code with changes mentioned.

**Comment by junaid-softaxus** [comment]
The endpoint is not returning response in correct format. Why it is in XML instead of JSON format?

File: src/CryptoAxus.API/Controllers/ArtistController.cs (hunk 5 above)
Comment (Afaqsoftaxus) [reply src/CryptoAxus.API/Controllers/ArtistController.cs:191]: Resolved

File: src/CryptoAxus.API/Controllers/RootController.cs (hunk 6 above)
Comment (Afaqsoftaxus) [reply src/CryptoAxus.API/Controllers/RootController.cs:39]: Resolved

File: src/CryptoAxus.API/Controllers/ArtistController.cs [hunk 15]
```diff
@@ -162,6 +187,13 @@ private IReadOnlyList<Links> CreateArtistLinks(ArtistDto dto, string? fields)
...
+                         Constants.PatchMethod);
```
[the remark this answers is not in the document]
Comment (Afaqsoftaxus) [reply src/CryptoAxus.API/Controllers/ArtistController.cs:192]: Resolved

File: src/CryptoAxus.API/Controllers/ArtistController.cs [hunk 16]
```diff
@@ -1,4 +1,6 @@
-﻿using CryptoAxus.Application.Features.Artist.PostArtist.Request;
+﻿using CryptoAxus.Application.Features.Artist.GetArtistByWalletAddress.Request;
+using CryptoAxus.Application.Features.Artist.GetArtistByWalletAddress.Response;
+using CryptoAxus.Application.Features.Artist.PostArtist.Request;
```
[the remark this answers is not in the document]
Comment (Afaqsoftaxus) [reply src/CryptoAxus.API/Controllers/ArtistController.cs:4]: Resolved

File: src/CryptoAxus.API/Controllers/ArtistController.cs (hunk 1 above)
Comment (Afaqsoftaxus) [reply src/CryptoAxus.API/Controllers/ArtistController.cs:126]: Resolved

File: src/CryptoAxus.API/Controllers/ArtistController.cs (hunk 2 above)
Comment (Afaqsoftaxus) [reply src/CryptoAxus.API/Controllers/ArtistController.cs:127]: Resolved

File: src/CryptoAxus.Application/Features/Artist/GetArtistByWalletAddress/Handler/GetArtistDataByWalletAddress.cs (hunk 8 above)
Comment (Afaqsoftaxus) [reply src/CryptoAxus.Application/Features/Artist/GetArtistByWalletAddress/Handler/GetArtistDataByWalletAddress.cs:16]: Resolved

File: src/CryptoAxus.Application/Features/Artist/GetArtistByWalletAddress/Handler/GetArtistDataByWalletAddress.cs [hunk 17]
```diff
@@ -0,0 +1,33 @@
...
+                                                      CancellationToken cancellationToken = default)
+    {
+        ArgumentNullException.ThrowIfNull(request, nameof(request));
+
+        ArtistDocument artist = await _repository.FindOneAsync(x => x.UserWalletAddress.Equals(request.UserWalletAddress));
+
+        if (artist is null)
+            return new GetArtistByWalletAddressResponse(HttpStatusCode.NotFound,
+                                             message: $"No artist found against WalletAddress: {request.UserWalletAddress}",
+                                             result: null);
```
[the remark this answers is not in the document]
Comment (Afaqsoftaxus) [reply src/CryptoAxus.Application/Features/Artist/GetArtistByWalletAddress/Handler/GetArtistDataByWalletAddress.cs:25]: Resolved

File: src/CryptoAxus.Application/Features/Artist/GetArtistByWalletAddress/Request/GetArtistByWalletAddressRequest.cs (hunk 10 above)
Comment (Afaqsoftaxus) [reply src/CryptoAxus.Application/Features/Artist/GetArtistByWalletAddress/Request/GetArtistByWalletAddressRequest.cs:1]: Resolved

File: src/CryptoAxus.Application/Features/Artist/GetArtistByWalletAddress/Response/NotFoundArtistByWalletAddressResponseExample.cs (hunk 11 above)
Comment (Afaqsoftaxus) [reply src/CryptoAxus.Application/Features/Artist/GetArtistByWalletAddress/Response/NotFoundArtistByWalletAddressResponseExample.cs:1]: Resolved

File: test/CryptoAxus.Application.Features.Artist/GetArtistByWalletAddress/Handler/GetArtistByWalletAddressHandlerTests.cs (hunk 13 above)
Comment (Afaqsoftaxus) [reply test/CryptoAxus.Application.Features.Artist/GetArtistByWalletAddress/Handler/GetArtistByWalletAddressHandlerTests.cs:6]: Resolved

File: test/CryptoAxus.Application.Features.Artist/GetArtistByWalletAddress/GetArtistByWalletAddressTestData.cs (hunk 12 above)
Comment (Afaqsoftaxus) [reply test/CryptoAxus.Application.Features.Artist/GetArtistByWalletAddress/GetArtistByWalletAddressTestData.cs:4]: Resolved

File: src/CryptoAxus.Application/Features/Artist/GetArtistByWalletAddress/Handler/GetArtistDataByWalletAddress.cs (hunk 9 above)
Comment (Afaqsoftaxus) [reply src/CryptoAxus.Application/Features/Artist/GetArtistByWalletAddress/Handler/GetArtistDataByWalletAddress.cs:31]: Resolved

File: src/CryptoAxus.Application/Features/Artist/GetArtistByWalletAddress/Handler/GetArtistDataByWalletAddress.cs (hunk 17 above)
[the remark this answers is not in the document]
Comment (Afaqsoftaxus) [reply src/CryptoAxus.Application/Features/Artist/GetArtistByWalletAddress/Handler/GetArtistDataByWalletAddress.cs:25]: Resolved

File: src/CryptoAxus.API/Controllers/ArtistController.cs (hunk 3 above)
Comment (Afaqsoftaxus) [reply src/CryptoAxus.API/Controllers/ArtistController.cs:132]: Resolved

File: src/CryptoAxus.API/Controllers/ArtistController.cs (hunk 4 above)
Comment (Afaqsoftaxus) [reply src/CryptoAxus.API/Controllers/ArtistController.cs:133]: Resolved

File: src/CryptoAxus.Application/Features/Artist/GetArtistByWalletAddress/Response/GetArtistByWalletAddressResponseExample.cs [hunk 18]
```diff
@@ -0,0 +1,9 @@
+﻿namespace CryptoAxus.Application.Features.Artist.GetArtistByWalletAddress.Response;
+
+public class GetArtistByWalletAddressResponseExample : IExamplesProvider<GetArtistByWalletAddressResponse>
+{
+    public GetArtistByWalletAddressResponse GetExamples()
+    {
+        return new GetArtistByWalletAddressResponse(HttpStatusCode.OK, "Artist record found successfully", new ArtistDto());
```
[the remark this answers is not in the document]
Comment (Afaqsoftaxus) [reply src/CryptoAxus.Application/Features/Artist/GetArtistByWalletAddress/Response/GetArtistByWalletAddressResponseExample.cs:7]: Resolved

File: src/CryptoAxus.Application/Features/Artist/GetArtistByWalletAddress/Handler/GetArtistDataByWalletAddress.cs (hunk 7 above)
Comment (junaid-softaxus) [reply src/CryptoAxus.Application/Features/Artist/GetArtistByWalletAddress/Handler/GetArtistDataByWalletAddress.cs:2]: Move the using statements in the relevant GlobalUsing file

File: src/CryptoAxus.Application/Features/Artist/GetArtistByWalletAddress/Handler/GetArtistDataByWalletAddress.cs [hunk 19]
```diff
@@ -0,0 +1,33 @@
...
+    {
+        _repository = repository;
+    }
+
+    public async Task<GetArtistByWalletAddressResponse> Handle(GetArtistByWalletAddressRequest request,
+                                                               CancellationToken cancellationToken = default)
+    {
+        ArgumentNullException.ThrowIfNull(request, nameof(request));
+
+        ArtistDocument artist = await _repository.FindOneAsync(x => x.UserWalletAddress.Equals( request.UserWalletAddress));
```
Comment (junaid-softaxus) [inline src/CryptoAxus.Application/Features/Artist/GetArtistByWalletAddress/Handler/GetArtistDataByWalletAddress.cs:21]: Remove space after `Equals(`

File: src/CryptoAxus.Application/Features/Artist/GetArtistByWalletAddress/Response/BadRequestArtistByWalletAddressResponseExample.cs [hunk 20]
```diff
@@ -0,0 +1,18 @@
...
+    {
+        return new BadRequestArtistByWalletAddressResponse
+        {
+            ApiException = null,
+            Links = null,
+            Errors = null,
+            IsSuccessful = false,
+            Message = "Unable to delete artist against userWalletAddress: 647115d2b38bc8ea242beb01",
+            Result = null,
+            StatusCode = HttpStatusCode.BadRequest
```
Comment (junaid-softaxus) [inline src/CryptoAxus.Application/Features/Artist/GetArtistByWalletAddress/Response/BadRequestArtistByWalletAddressResponseExample.cs:15]: Please view the ticket details as the Bad request example provided there indicates that the response message should be **Invalid media type provided**

File: src/CryptoAxus.Application/Features/Artist/GetArtistByWalletAddress/Handler/GetArtistDataByWalletAddress.cs (hunk 7 above)
Comment (Afaqsoftaxus) [reply src/CryptoAxus.Application/Features/Artist/GetArtistByWalletAddress/Handler/GetArtistDataByWalletAddress.cs:2]: Resolved

File: src/CryptoAxus.Application/Features/Artist/GetArtistByWalletAddress/Handler/GetArtistDataByWalletAddress.cs (hunk 19 above)
Comment (Afaqsoftaxus) [reply src/CryptoAxus.Application/Features/Artist/GetArtistByWalletAddress/Handler/GetArtistDataByWalletAddress.cs:21]: Resolved

File: src/CryptoAxus.Application/Features/Artist/GetArtistByWalletAddress/Response/BadRequestArtistByWalletAddressResponseExample.cs (hunk 20 above)
Comment (Afaqsoftaxus) [reply src/CryptoAxus.Application/Features/Artist/GetArtistByWalletAddress/Response/BadRequestArtistByWalletAddressResponseExample.cs:15]: Resolved

File: src/CryptoAxus.API/Controllers/ArtistController.cs [hunk 21]
```diff
@@ -123,6 +121,32 @@ public async Task<IActionResult> PostArtist([FromBody] CreateArtistDto artist)
...
+    [HttpGet("{userWalletAddress:alpha:required}/userWalletAddress", Name = "GetArtistByWalletAddressRequest", Order = 5)]
+    [RequiresParameter(Name = "userWalletAddress", Required = true, Source = OpenApiParameterLocation.Path, Type = typeof(string))]
+    [SwaggerRequestExample(typeof(GetArtistByWalletAddressRequest), typeof(GetArtistByWalletAddressRequestExample))]
+    [ProducesResponseType(typeof(GetArtistByWalletAddressResponse), (int)HttpStatusCode.OK)]
+    [ProducesResponseType(typeof(NotFoundArtistByWalletAddressResponse), (int)HttpStatusCode.NotFound)]
+    [ProducesResponseType(typeof(BadRequestArtistByWalletAddressResponse), (int)HttpStatusCode.BadRequest)]
+    public async Task<IActionResult> GetArtistByWalletAddressRequest([FromRoute] string userWalletAddress)
+    {
+        var response = await Mediator.Send(new GetArtistByWalletAddressRequest(userWalletAddress));
+
```
Comment (junaid-softaxus) [inline src/CryptoAxus.API/Controllers/ArtistController.cs:141]: Why media type not provided as part of the query string and implementation for parsing the media type value not added. See `GetArtistById` method.

**Review by junaid-softaxus — COMMENTED** [review]
Feedback added

File: src/CryptoAxus.API/Controllers/ArtistController.cs (hunk 21 above)
Comment (Afaqsoftaxus) [reply src/CryptoAxus.API/Controllers/ArtistController.cs:141]: Resolved

**Review by junaid-softaxus — COMMENTED** [review]
Reviewed

**Pull request merged**